mm2src/mm2_main/src/lp_wallet.rs

```diff
@@ -0,0 +1,504 @@
+use common::HttpStatusCode;
+use crypto::{decrypt_mnemonic, encrypt_mnemonic, generate_mnemonic, CryptoCtx, CryptoInitError, EncryptedData,
+             MnemonicError};
+use http::StatusCode;
+use mm2_core::mm_ctx::MmArc;
+use mm2_err_handle::prelude::*;
+use serde::de::DeserializeOwned;
+use serde_json::{self as json};
+
+cfg_wasm32! {
+    use crate::mm2::lp_wallet::mnemonics_wasm_db::{WalletsDb, WalletsDBError};
+    use mm2_core::mm_ctx::from_ctx;
+    use mm2_db::indexed_db::{ConstructibleDb, DbLocked, InitDbResult};
+    use mnemonics_wasm_db::{read_encrypted_passphrase_if_available, save_encrypted_passphrase};
+    use std::sync::Arc;
+
+    type WalletsDbLocked<'a> = DbLocked<'a, WalletsDb>;
+}
+
+cfg_native! {
+    use mnemonics_storage::{read_encrypted_passphrase_if_available, save_encrypted_passphrase, WalletsStorageError};
+}
+
+#[cfg(not(target_arch = "wasm32"))]
+#[path = "lp_wallet/mnemonics_storage.rs"]
+mod mnemonics_storage;
+#[cfg(target_arch = "wasm32")]
+#[path = "lp_wallet/mnemonics_wasm_db.rs"]
+mod mnemonics_wasm_db;
+
+type WalletInitResult<T> = Result<T, MmError<WalletInitError>>;
+
+#[derive(Debug, Deserialize, Display, Serialize)]
+pub enum WalletInitError {
+    #[display(fmt = "Error deserializing '{}' config field: {}", field, error)]
+    ErrorDeserializingConfig {
+        field: String,
+        error: String,
+    },
+    #[display(fmt = "The '{}' field not found in the config", field)]
+    FieldNotFoundInConfig {
+        field: String,
+    },
+    #[display(fmt = "Wallets storage error: {}", _0)]
+    WalletsStorageError(String),
+    #[display(
+        fmt = "Passphrase doesn't match the one from file, please create a new wallet if you want to use a new passphrase"
+    )]
+    PassphraseMismatch,
+    #[display(fmt = "Error generating or decrypting mnemonic: {}", _0)]
+    MnemonicError(String),
+    #[display(fmt = "Error initializing crypto context: {}", _0)]
+    CryptoInitError(String),
+    Internal(String),
```

Comment:
Maybe good to name internal errors everywhere identically ('InternalError')

Reply:
Done